### DISCUSSION TITLE
Copilot generated options selection with arrow keys blocks navigation

### DISCUSSION THREAD

**Discussion by scotteverett** [community/community#54938] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

The copilot arrow selection of generated options ends up working against a natural coding flow more than it benefits.  It blocks simple navigation interactions, needing to be escaped before being able to continue with the coding flow.   I find It reduces efficiency and devalues the copilot feature significantly, especially when editing and the generated suggestions are not relevant.

**Comment by Akash1134** [comment]
Hi @scotteverett ,

Thanks for sharing your feedback, that means a lot. Regarding the issue you've highlighted above, as with most generative AI products, Copilot also makes its best guess based on the statistics and is subjected to making a few mistakes.

When GitHub Copilot synthesizes code suggestions based on available public data, it can also synthesize code that contains some undesirable patterns. This is something we care a lot about at GitHub, and in recent years we’ve provided tools such as GitHub Actions, Dependabot, and CodeQL to open-source projects to help improve code quality. Of course, you should always use GitHub Copilot together with good testing and code review practices and security tools, as well as your own judgment.

It would be great if you could share some example scenarios where you feel that Copilot did not do a great job. I'll share them with our engineers to look into. 🙂 

I'd recommend reading these [FAQs](https://github.com/features/copilot#:~:text=Frequently%20asked%C2%A0questions) for more details about how Copilot suggests code and how you can leverage it at its best.

**Comment by scotteverett** [comment]
Hi Akash,

Thanks for responding.  The specific behaviour that I'm referencing is the
introduction of the small feature of using the arrow keys <> to cycle
through inline copilot suggestions.  When an edit is suggested in grey but
you don't want to select it using tab you can't use the right arrow key to
navigate until you either hit escape or up.   When you're trying to rapidly
edit code, I find this small behaviour ends up being quite disruptive.  I
really like copilot a lot and rely on it heavily in framing in code blocks,
I've been using it since its release, so I have really noticed this small
feature change in the last few weeks.  It feels less efficient than the
suggestion selection behaviour in earlier versions.

Best,

Scott

**Reply by Akash1134** [reply]
Thanks for the update Scott. I have shared this with our team internally. 🙂

**Reply by prkgnt** [reply]
Hi :) I'm facing the same issue. Is there any update about this issue? Still I can't navigate using arrow key when suggestion showed up :(
